src/test/java/com/firebolt/jdbc/client/UsageTrackerUtilTest.java

```diff
@@ -185,35 +160,16 @@ void shouldParseDetectStack() {
 		}
 	}
 
-	@Test
-	void shouldNotOverrideConnectorsIf100NameVersionPairsAreSpecified() {
-		System.setProperty("os.name", "MacosX");
-		Map<String, String> connectors = new HashMap<>();
-		StringBuilder connectorsInfo = new StringBuilder("ConnB:2.0.1");
-		for (int i = 0; i < 99; i++) {
-			connectorsInfo.append(",ConnA:1.1.1");
-		}
-		connectors.put("ConnA", "1.2.0");
-		connectors.put("ConnB", "3.0.4");
-		try (MockedStatic<UsageTrackerUtil> mock = mockClients(new HashMap<>(), connectors)) {
-			String result = UsageTrackerUtil.getUserAgentString("", connectorsInfo.toString());
-			assertEquals("ConnA/1.1.1 ConnB/2.0.1 JDBC/1.0-TEST (Java 8.0.1; Darwin 10.1; )", result);
-		}
-	}
-
-	@Test
-	void shouldNotOverrideConnectorsIfMoreThan100NameVersionPairsAreSpecified() {
-		System.setProperty("os.name", "MacosX");
-		Map<String, String> connectors = new HashMap<>();
-		StringBuilder connectorsInfo = new StringBuilder("ConnB:2.0.1");
-		for (int i = 0; i < 100; i++) {
-			connectorsInfo.append(",ConnA:1.1.1");
-		}
-		connectors.put("ConnA", "1.2.0");
-		connectors.put("ConnB", "3.0.4");
-		try (MockedStatic<UsageTrackerUtil> mock = mockClients(new HashMap<>(), connectors)) {
-			String result = UsageTrackerUtil.getUserAgentString("", connectorsInfo.toString());
-			assertEquals("ConnA/1.2.0 ConnB/3.0.4 JDBC/1.0-TEST (Java 8.0.1; Darwin 10.1; )", result);
-		}
+	@ParameterizedTest(name = "{1}")
+	@CsvSource(value = {
+			"98,less than 100,ConnA/1.1.1 ConnB/2.0.1",
+			"99,equals to 100,ConnA/1.1.1 ConnB/2.0.1",
+			"100, greater than 100,ConnA/1.2.0 ConnB/3.0.4",
+			"101, seriously greater than 100,ConnA/1.2.0 ConnB/3.0.4"
+	}, delimiter = ',')
+	void shouldNotOverrideConnectorsIfMoreThanSeveralNameVersionPairsAreSpecified(int n, String name, String expectedUserAgentPrefix) {
```

Comment:
Is `name` used anywhere or is it added for readability?

Reply:
yes, it is added for readability for similarity with the previous implementation where each such test was implemented in  separate function. "less than 100" is clearer than 99